subsys/logging/log_core.c

```diff
@@ -44,9 +49,20 @@ static inline void msg_finalize(struct log_msg *msg,
 	}
 }
 
+static inline void log_cond_init(void)
+{
+	if (!initialized) {
```

Comment:
As the logging could be called from multiple threads that could be preempted, shouldn't the interface variable be atomic so that we would not be able call log_init() twice. By looking the code, it seems possible that log_init() could be initialized more than once.

Reply:
@jukkar good point, I've added following mechanism: logger can have 3 internal states: uninit'ed, init in progress, init'ed. each log API is atomic_inc state and only if previous state was uninit'ed it will perform initialization, if state is init in progress then log is dropped because we cannot do much since we interrupt log initialization.

I've added kindof state initial precheck before doing atomic incrementation for performance reasons.